### PR TITLE
New component & container pages

### PR DESCRIPTION
**Pull request by jwmza**
This adds a page to the GUI for creating a new container
image

as well as:

- Adds a **New Component** screen to select which type of component you'd like to add to a workspace
- Cleans up existing creation forms and updates the overall design&layout
- Trims `:out` and `:err` from logs in the GUI, similar to the command line log output.

### CODE REVIEW

**Comment by jwmza** [comment]
Well, this is working for create
image

File: gui/src/components/logs/Logs.svelte [hunk 1]
```diff
@@ -47,7 +47,9 @@
         >
           {event.time.short}
         </td>
-        <td class="name">{event.name}</td>
+        <td class="name" title={event.name}
+          >{event.name.replace(/(:err$)|(:out$)/g, '')}</td
```
Comment (brandonbloom) [inline gui/src/components/logs/Logs.svelte:51]: Please tag this with a `SEE NOTE [LOG_COMPONENTS]` command, as this is a problem spread across many places in the code.

**Comment by jwmza** [comment]
This is still very much WIP btw, but the Monaco editor *can* be used to create a container.

File: gui/src/routes.ts [hunk 2]
```diff
@@ -18,7 +20,9 @@ export default {
   '/workspaces/:workspace/components': WorkspaceComponents,
   '/workspaces/:workspace/storage': WorkspaceStorage,
   '/workspaces/:workspace/networking': WorkspaceNetworking,
+  '/workspaces/:workspace/new-component': WorkspaceNewComponent,
   '/workspaces/:workspace/new-process': NewProcess,
+  '/workspaces/:workspace/new-docker-container': NewDockerContainer,
```
Comment (brandonbloom) [inline gui/src/routes.ts:25]: let's just call this "new-container" and drop the word "docker". We may still want to refer to the category of things as "docker" or the provider as such, but as a general rule containers etc are outgrowing docker-the-brand.

**Comment by brandonbloom** [comment]
Since this is still a WIP/Draft, didn't do a thorough review, but looking like its on the right track! If you have some bits working, let's get it polished up so we can land work on master incrementally.

**Comment by jwmza** [comment]
image

File: gui/src/components/form/EditAs.svelte [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+  export let mode: string;
+  export let editorModes: Record<string, any>[];
```
Comment (brandonbloom) [inline gui/src/components/form/EditAs.svelte]: better type than `any`

File: gui/src/pages/NewProcess.svelte [hunk 4]
```diff
@@ -49,6 +50,22 @@
     });
   };
 
+  let mode = 'script';
+  const editorModes = [
+    {
+      id: 'script',
+      name: 'Script',
+      effect: updateScript,
```
Comment (brandonbloom) [inline gui/src/pages/NewProcess.svelte]: let's rename this to something clearer with an `on` prefix. For example: `onActivate`

File: gui/src/components/ProcessList.svelte [hunk 5]
```diff
@@ -63,7 +63,7 @@
       tooltip="Add new process"
```
Comment (brandonbloom) [inline gui/src/components/ProcessList.svelte:63]:
```suggestion
      tooltip="Add component"
```

File: gui/src/lib/api.ts [hunk 6]
```diff
@@ -211,6 +217,12 @@ export interface WorkspaceApi {
     spec: ProcessSpec,
   ): Promise<CreateProcessResponse>;
 
+  createComponent(
```
Comment (brandonbloom) [inline gui/src/lib/api.ts:220]: we should probably use this in favor over createProcess everywhere

File: gui/src/pages/WorkspaceNewComponent.svelte [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    title="New component"
+    backRoute={workspaceRoute}
+    --panel-padding="2rem"
+    --panel-overflow-y="scroll"
+  >
+    <div class="center-form">
+      <section>
+        <!-- Generic components, no heading. -->
+
+        <button
```
Comment (brandonbloom) [inline gui/src/pages/WorkspaceNewComponent.svelte:27]: TODO: `LinkButton` component

File: gui/src/pages/WorkspaceNewComponent.svelte [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            router.push(
+              `/workspaces/${encodeURIComponent(workspaceId)}/new-container`,
+            );
+          }}
+        >
+          <DockerSVG />
+          <b>Container</b>
+        </button>
+
+        <!-- Volume, network, etc. -->
```
Comment (brandonbloom) [inline gui/src/pages/WorkspaceNewComponent.svelte:55]: Fast follow: Volume & Network.

File: gui/src/pages/NewDockerContainer.svelte [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+  export let params = { workspace: '' };
+
+  const workspaceId = params.workspace;
+  const workspace = api.workspace(workspaceId);
+  const workspaceRoute = `/workspaces/${encodeURIComponent(workspaceId)}`;
+  const workspaceNewComponentRoute = `/workspaces/${encodeURIComponent(
+    workspaceId,
+  )}/new-component`;
+
+  const resetToLastSaved = () => {};
```
Comment (brandonbloom) [inline gui/src/pages/NewDockerContainer.svelte]: cut this for now

File: gui/src/pages/NewDockerContainer.svelte [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+        tooltip="Reset to last saved state"
+        on:click={resetToLastSaved}
+      >
+        <ResetSVG />
+      </IconButton>
+    </div>
+    <div class="center-form">
+      <h1><DockerSVG /> New Container</h1>
+      <form
+        on:submit|preventDefault={async () => {
```
Comment (brandonbloom) [inline gui/src/pages/NewDockerContainer.svelte:57]: next step: make this reusable between all component create flows

**Review by brandonbloom — APPROVED** [review]

**Pull request merged**